### PR TITLE
Upgrade wp-prettier to 2.8.5

### PR DESCRIPTION
**Pull request by jsnajdr**
Updates the `wp-prettier` package to version 2.8.5 and reformats the codebase.

Notable changes:
- fixes extra space in ternary expressions (https://github.com/Automattic/wp-prettier/issues/34, thx @anomiex for reporting)
- type expressions like `typeof props[ 'value' ]` now get parens for clarity: `( typeof props )[ 'value' ]`
- support for latest TypeScript 5.0 syntax: `satisfies`, `in` and `out` variance specifiers, ...

### CODE REVIEW

**Review by aristath — APPROVED** [review]
LGTM 👍

File: packages/block-editor/src/hooks/content-lock-ui.js [hunk 1]
```diff
@@ -104,9 +104,13 @@ export const withBlockControls = createHigherOrderComponent(
 			return <BlockEdit { ...props } />;
 		}
 
+		const showStopEditingAsBlocks = isEditingAsBlocks && ! isContentLocked;
+		const showStartEditingAsBlocks =
+			! isEditingAsBlocks && isContentLocked && props.isSelected;
```
Comment (tyxla) [inline packages/block-editor/src/hooks/content-lock-ui.js:109]: I'm curious why those were needed as separate variables. Is the new version breaking these expressions somehow?

File: packages/components/src/border-box-control/stories/index.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ const colors = [
 
 const Template: ComponentStory< typeof BorderBoxControl > = ( props ) => {
 	const { onChange, ...otherProps } = props;
-	const [ borders, setBorders ] = useState< typeof props[ 'value' ] >();
+	const [ borders, setBorders ] = useState< ( typeof props )[ 'value' ] >();
```
Comment (tyxla) [inline packages/components/src/border-box-control/stories/index.tsx:46]: This does make sense IMO 👍

File: packages/edit-post/src/store/actions.js [hunk 3]
```diff
@@ -294,7 +294,7 @@ export const showBlockTypes =
 			( type ) =>
 				! (
 					Array.isArray( blockNames ) ? blockNames : [ blockNames ]
-				 ).includes( type )
+				).includes( type )
```
Comment (tyxla) [inline packages/edit-post/src/store/actions.js:297]: Oh, I love this. I hated how it added that extra space!

**Review by tyxla — APPROVED** [review]
Thanks, @jsnajdr! Great work 💯 

This version looks great and is obviously an improvement (especially the removed extra space). 

I had a question, but it's non-blocking one, feel free to 🚢 

🚀

File: packages/block-editor/src/hooks/content-lock-ui.js (hunk 1 above)
Comment (jsnajdr) [reply packages/block-editor/src/hooks/content-lock-ui.js:109]: Not breaking them, the formatting was just not pretty. The conditional JSX:
```jsx
<>
  { isEditing && isLocked && isSelected && (
    <BlockSettings />
  ) }
</>
```
was already over 80 columns, but the old Prettier didn't break it. The new Prettier breaks it like:
```jsx
<>
  { isEditing &&
    isLocked &&
    isSelected && (
      <BlockSettings />
    ) }
</>
```
which looks bad and also adds a new level of indentation for the inner markup. I decided to make it look good again by extracting the condition.

**Review by gziolo — APPROVED** [review]
Excellent, so great to see it upgraded 👍

File: packages/block-editor/src/hooks/content-lock-ui.js (hunk 1 above)
Comment (tyxla) [reply packages/block-editor/src/hooks/content-lock-ui.js:109]: Agreed, thank you for clarifying 👍

**Pull request merged**